tests/function_executor/test_cloud_events.py

```diff
@@ -0,0 +1,371 @@
+import io
+import json
+import sys
+import unittest
+from datetime import datetime, timezone
+
+from pydantic import BaseModel
+
+from tensorlake.applications import SerializationError
+from tensorlake.function_executor.cloud_events import (
+    new_cloud_event,
+    print_cloud_event,
+)
+
+
+class TestCloudEventStructure(unittest.TestCase):
+    """Tests for CloudEvent structure and CloudEvents 1.0 spec compliance."""
+
+    def test_new_cloud_event_creates_valid_structure(self):
+        """Test that new_cloud_event creates a valid CloudEvent structure."""
+        event_data = {"test": "data"}
+        event = new_cloud_event(event_data)
+
+        # Verify required CloudEvents 1.0 spec fields
+        self.assertIn("specversion", event)
+        self.assertEqual(event["specversion"], "1.0")
+        self.assertIn("id", event)
+        self.assertIn("timestamp", event)
+        self.assertIn("type", event)
+        self.assertIn("source", event)
+        self.assertIn("data", event)
+
+        # Verify data is preserved
+        self.assertEqual(event["data"], event_data)
+
+    def test_new_cloud_event_default_type(self):
```

Comment:
I'm okay to keep these unit tests but many of them seem to be excessive to me. We'll have to change them with every minor not important change.

Reply:
I've removed some tests from there that were not critical.